auth/tests/api/users.rs

```diff
@@ -227,6 +229,74 @@ mod needs_docker {
         );
     }
 
+    #[tokio::test]
+    async fn update_subscription_endpoint_requires_jwt() {
+        let app = app().await;
+
+        let subscription_item = serde_json::to_string(&SubscriptionItem::new(
+            shuttle_common::backends::subscription::PriceId::AwsRdsRecurring,
+            1,
+        ))
+        .unwrap();
+
+        // POST /users/subscription/items without bearer JWT.
+        let request = Request::builder()
+            .uri("/users/subscription/items")
+            .method("POST")
+            .header(CONTENT_TYPE, "application/json")
+            .body(Body::from(subscription_item.clone()))
+            .unwrap();
+
+        let response = app.send_request(request).await;
+
+        assert_eq!(response.status(), StatusCode::INTERNAL_SERVER_ERROR);
```

Comment:
The layer is designed to return 400 (although it should be 401) for this, but since there is no token in the request axum returns a 500 due to failing to set the extension. When we pass an invalid token in the next test, however, the extension should not be set either, but that returns the expected 400. :thinking:

Reply:
> but that returns the expected 400.

You mean 401?

Should we open a ticket for this bug if it is unrelated to the code changes, and fix it separately? It is important for our monitoring. @oddgrd you can assign it to me if you do it.

Reply:
Expected in the sense that the current jwt layer implementation would return a 400 for a missing bearer token. I'll open a ticket for it, since I am a bit confused about the code path that leads to this behavior: https://github.com/shuttle-hq/shuttle/blob/main/common/src/backends/auth.rs#L351-L355.

Reply:
It looks like we just proceed with the request, calling inner if there is no bearer token, which seems incorrect since that should return a 401.

Reply:
This comment https://github.com/shuttle-hq/shuttle/pull/1478#discussion_r1430498392 prompted me to add a scopedlayer to the endpoint, and now it does return 401, since the jwt auth layer calls the next layer, which is the scopedlayer, and that correctly returns 401 for missing token.